src/utils/schema.ts

```diff
@@ -236,3 +205,45 @@ export function encodeProxyCall(address: string, howToCall: HowToCall, calldata:
   const abi = shouldAssert ? proxyAssertABI : proxyABI
   return encodeCall(abi, [address, howToCall, Buffer.from(calldata.slice(2), 'hex')])
 }
+
+// Helpers for atomicizer
+
+function encodeAtomicizedCalldata(atomicizer: WyvernAtomicizerContract, schemas: Array<Schema<WyvernAsset>>, assets: WyvernAsset[], address: string, side: OrderSide) {
+
+  const encoder = side === OrderSide.Sell ? encodeSell : encodeBuy
+
+  try {
+    const transactions = assets.map((asset, i) => {
+      const schema = schemas[i]
+      const { target, calldata } = encoder(schema, asset, address)
+      return {
+        calldata,
+        abi: schema.functions.transfer(asset),
+        address: target,
+        value: new BigNumber(0),
+      }
+    })
+
+    const atomicizedCalldata = atomicizer.atomicize.getABIEncodedTransactionData(
+      transactions.map(t => t.address),
+      transactions.map(t => t.value),
+      transactions.map(t => new BigNumber((t.calldata.length - 2) / 2)), // subtract 2 for '0x', divide by 2 for hex
+      transactions.map(t => t.calldata).reduce((x, y) => x + y.slice(2)), // cut off the '0x'
+    )
+
+    const kind = side === OrderSide.Buy ? FunctionInputKind.Owner : undefined
+
+    const atomicizedReplacementPattern = WyvernProtocol.encodeAtomicizedReplacementPattern(transactions.map(t => t.abi), kind)
+
+    if (!atomicizedCalldata || !atomicizedReplacementPattern) {
+      throw new Error(`Invalid calldata: ${atomicizedCalldata}, ${atomicizedReplacementPattern}`)
+    }
+    return {
+      atomicizedCalldata,
+      atomicizedReplacementPattern
+    }
+  } catch (error) {
+    console.error({ schemas, assets, address, side })
+    throw new Error(`Failed to construct your order: likely something strange about this type of item. OpenSea has been notified. Please contact us in Discord! Original error: ${error}`)
```

Comment:
@DJViau this is the new part that handles errors for constructing bundle calldata